vertx-pojo-mapper-common/src/main/java/de/braintags/io/vertx/pojomapper/dataaccess/query/IQuery.java

```diff
@@ -31,46 +31,64 @@
 public interface IQuery<T> extends IDataAccessObject<T> {
 
   /**
+   * Execute the query. Any variables in the search condition will result in an error. For queries with variables, see
+   * {@link #execute(IFieldValueResolver, int, int, Handler)}
+   *
+   * @param resultHandler
+   */
+  void execute(Handler<AsyncResult<IQueryResult<T>>> resultHandler);
+
+  /**
    * Execute the query
    *
+   * @param resolver
+   *          replaces potential variables in the search condition with an actual value, can be null
+   * @param limit
+   *          the maximum number of results to search
+   * @param offset
+   *          the offset of the first row to return
    * @param resultHandler
    *          contains the {@link IQueryResult}
    */
-  void execute(Handler<AsyncResult<IQueryResult<T>>> resultHandler);
+  void execute(IFieldValueResolver resolver, int limit, int offset,
+      Handler<AsyncResult<IQueryResult<T>>> resultHandler);
 
   /**
-   * Execute the query by counting the fitting objects
+   * Execute the query by counting the fitting objects. Any variables in the search condition will result in an error.
+   * For queries with variables, see {@link #executeCount(IFieldValueResolver, Handler)}
    *
    * @param resultHandler
-   *          contains the {@link IQueryCountResult}
    */
   void executeCount(Handler<AsyncResult<IQueryCountResult>> resultHandler);
 
   /**
-   * Execute the query with the option explain and sends back the suitable information
+   * Execute the query by counting the fitting objects
    *
+   * @param resolver
+   *          replaces potential variables in the search condition with an actual value, can be null
    * @param resultHandler
-   *          contains the {@link IQueryResult}
+   *          contains the {@link IQueryCountResult}
    */
-  void executeExplain(Handler<AsyncResult<IQueryResult<T>>> resultHandler);
+  void executeCount(IFieldValueResolver resolver, Handler<AsyncResult<IQueryCountResult>> resultHandler);
 
   /**
-   * Set the maximum number of records to be returned
+   * Execute the query with the option explain and sends back the suitable information
    *
-   * @param limit
-   *          the maximum number of records to be returned
-   * @return the query itself for fluent access
+   * @param resultHandler
+   *          contains the {@link IQueryResult}
    */
-  IQuery<T> setLimit(int limit);
```

Comment:
It's given as parameter to the execution directly:
void execute(IFieldValueResolver resolver, int limit, int offset,
      Handler<AsyncResult<IQueryResult<T>>> resultHandler);

Query still has another method without limit that uses the new default value of the datastore.